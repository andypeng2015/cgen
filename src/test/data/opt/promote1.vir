module foo

export function w $foo(w %x) {
  %a = slot 4, align 4
@start:
  st.w %x, %a
  %y = ld.w %a
  %y = add.w %y, %x
  st.w %y, %a
  %y = ld.w %a
  ret %y
}
