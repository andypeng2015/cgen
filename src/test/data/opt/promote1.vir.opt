module foo

export function w $foo(w %x) {
@2:
  %1 = add.w %x, %x ; @9
  ret %1
}
