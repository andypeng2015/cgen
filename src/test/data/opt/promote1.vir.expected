module foo

export function w $foo(w %x) {
@2:
  %0 = add.w %x, %x ; @8
  ret %0
}
